app/api/promptset/route.ts

```diff
@@ -1,124 +1,204 @@
 import { authOptions } from "@/lib/auth/options";
 import prisma from "@/lib/prisma";
+import { authApiKey, fillPromptStringTemplate, parseQueryString } from "@/lib/utils";
 import { getServerSession } from "next-auth";
 import { redirect } from "next/navigation";
 import { NextRequest, NextResponse } from "next/server";
 
 export async function GET(req: NextRequest) {
   try {
-    const session = await getServerSession(authOptions);
-    if (!session || !session.user) {
-      redirect("/login");
```

Comment:
can you revert this back to redirect("/login"). we need to find a better way to do this but for now this is the approach we are taking for apis used by the client